### PR TITLE
Update avl_tree.py

### PR DESCRIPTION
**Pull request by zky001**
it's true definition of AVL tree,change  left and right rotation,and add avl_tree doctest

### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by zky001** [comment]
> Hey @zky001,
> Something went wrong with the build.
> 
> TravisCI finished with status `errored`, which means the build failed because of something unrelated to the tests, such as a problem with a dependency or the build process itself.
> 
> [View build log](https://travis-ci.com/TheAlgorithms/Python/builds/172418930)
> 
> ###### TravisBuddy Request Identifier: f92a6e90-b437-11ea-bd89-f57d85bb28d1

This error is caused by flake8 W291 trailing whitespace，this whitespace is caused by my doctest's output

**Comment by zky001** [comment]
> ## Travis tests have failed
> Hey @zky001,
> Please read the following log in order to understand the failure reason.
> It'll be awesome if you fix what's wrong and commit the changes.
> 
> ###### TravisBuddy Request Identifier: 4af64640-b43d-11ea-bd89-f57d85bb28d1

when I delete  this whitespace in my doctest, this action will raise test failed error.

**Comment by cclauss** [comment]
print(" ".join(line.strip() for line in my_doctest()))

File: data_structures/binary_tree/avl_tree.py [hunk 1]
```diff
@@ -1,5 +1,12 @@
+
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:1]:
```suggestion
```

File: data_structures/binary_tree/avl_tree.py [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
 """
-An auto-balanced binary tree!
+This is a pure Python implementation of An auto-balanced binary tree!
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:3]:
```suggestion
Implementation of an auto-balanced binary tree!
```

File: data_structures/binary_tree/avl_tree.py [hunk 3]
```diff
@@ -1,5 +1,12 @@
...
+For doctests run following command:
+python -m doctest -v avl_tree.py
+or
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:6]:
```suggestion
```

File: data_structures/binary_tree/avl_tree.py [hunk 4]
```diff
@@ -80,7 +87,7 @@ def my_max(a, b):
     return b
 
 
-def leftrotation(node):
+def rightrotation(node):
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:90]:
```suggestion
def right_rotation(node):
```

File: data_structures/binary_tree/avl_tree.py [hunk 5]
```diff
@@ -103,7 +109,7 @@ def leftrotation(node):
     return ret
 
 
-def rightrotation(node):
+def leftrotation(node):
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:112]:
```suggestion
def left_rotation(node):
```

File: data_structures/binary_tree/avl_tree.py [hunk 6]
```diff
@@ -118,24 +124,24 @@ def rightrotation(node):
     return ret
 
 
-def rlrotation(node):
+def lrrotation(node):
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:127]:
```suggestion
def lr_rotation(node):
```

File: data_structures/binary_tree/avl_tree.py [hunk 7]
```diff
@@ -118,24 +124,24 @@ def rightrotation(node):
...
          / \       -->  /  \         -->    /     / \
         Bl  Br         B   UB              Bl    UB  C
              \        /
              UB     Bl
     RR = rightrotation   LR = leftrotation
     """
-    node.setleft(rightrotation(node.getleft()))
-    return leftrotation(node)
+    node.setleft(leftrotation(node.getleft()))
+    return rightrotation(node)
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:139]:
```suggestion
    node.setleft(left_rotation(node.get_left()))
    return right_rotation(node)
```

File: data_structures/binary_tree/avl_tree.py [hunk 8]
```diff
@@ -118,24 +124,24 @@ def rightrotation(node):
...
 
 
-def lrrotation(node):
-    node.setright(leftrotation(node.getright()))
-    return rightrotation(node)
+def rlrotation(node):
+    node.setright(rightrotation(node.getright()))
+    return leftrotation(node)
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:144]:
```suggestion
def rl_rotation(node):
    node.set_right(right_rotation(node.get_right()))
    return left_rotation(node)
```

File: data_structures/binary_tree/avl_tree.py [hunk 9]
```diff
@@ -149,16 +155,16 @@ def insert_node(node, data):
             if (
                 data < node.getleft().getdata()
             ):  # new node is the left child of the left child
-                node = leftrotation(node)
+                node = rightrotation(node)
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:158]:
```suggestion
                node = right_rotation(node)
```

File: data_structures/binary_tree/avl_tree.py [hunk 10]
```diff
@@ -149,16 +155,16 @@ def insert_node(node, data):
...
             else:
-                node = rlrotation(node)  # new node is the right child of the left child
+                node = lrrotation(node)  # new node is the right child of the left child
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:160]:
```suggestion
                node = lr_rotation(node)  # new node is the right child of the left child
```

File: data_structures/binary_tree/avl_tree.py [hunk 11]
```diff
@@ -149,16 +155,16 @@ def insert_node(node, data):
...
     else:
         node.setright(insert_node(node.getright(), data))
         if getheight(node.getright()) - getheight(node.getleft()) == 2:
             if data < node.getright().getdata():
-                node = lrrotation(node)
+                node = rlrotation(node)
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:165]:
```suggestion
                node = rl_rotation(node)
```

File: data_structures/binary_tree/avl_tree.py [hunk 12]
```diff
@@ -149,16 +155,16 @@ def insert_node(node, data):
...
             else:
-                node = rightrotation(node)
+                node = leftrotation(node)
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:167]:
```suggestion
                node = left_rotation(node)
```

File: data_structures/binary_tree/avl_tree.py [hunk 13]
```diff
@@ -201,20 +207,52 @@ def del_node(root, data):
         return root
     if getheight(root.getright()) - getheight(root.getleft()) == 2:
         if getheight(root.getright().getright()) > getheight(root.getright().getleft()):
-            root = rightrotation(root)
+            root = leftrotation(root)
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:210]:
```suggestion
            root = left_rotation(root)
```

File: data_structures/binary_tree/avl_tree.py [hunk 14]
```diff
@@ -201,20 +207,52 @@ def del_node(root, data):
...
         else:
-            root = lrrotation(root)
+            root = rlrotation(root)
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:212]:
```suggestion
            root = rl_rotation(root)
```

File: data_structures/binary_tree/avl_tree.py [hunk 15]
```diff
@@ -201,20 +207,52 @@ def del_node(root, data):
...
     elif getheight(root.getright()) - getheight(root.getleft()) == -2:
         if getheight(root.getleft().getleft()) > getheight(root.getleft().getright()):
-            root = leftrotation(root)
+            root = rightrotation(root)
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:215]:
```suggestion
            root = right_rotation(root)
```

File: data_structures/binary_tree/avl_tree.py [hunk 16]
```diff
@@ -201,20 +207,52 @@ def del_node(root, data):
...
         else:
-            root = rlrotation(root)
+            root = lrrotation(root)
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:217]:
```suggestion
            root = lr_rotation(root)
```

File: data_structures/binary_tree/avl_tree.py [hunk 17]
```diff
@@ -201,20 +207,52 @@ def del_node(root, data):
...
 
 
 class AVLtree:
+    """
+    An AVL tree doctest
+    Examples:
+    >>> t = AVLtree()
+    >>> t.insert(4)
+    insert:4
+    >>> t.traversale()
```
Comment (cclauss) [inline data_structures/binary_tree/avl_tree.py:230]: Let's change the method `traversale()` --> `__str__()` and then change instances of `t.traversale()` into `str(t)`.

**Comment by cclauss** [comment]
Also, please change `isEmpty()` to `is_empty()`.

**Comment by zky001** [comment]
> Also, please change `isEmpty()` to `is_empty()`.

thanks a lot

**Pull request merged**